File: apps/macos/PasteSheets/Domain/Entities/PasteItem.swift
import Foundation

struct PasteItem: Identifiable, Equatable {
    let id: Int64
    let content: String
    let directory: String
    let createdAt: String
    let memo: String?

    init(dto: PasteItemDTO) {
        self.id = dto.id
        self.content = dto.content
        self.directory = dto.directory
        self.createdAt = dto.createdAt
        self.memo = dto.memo
    }
}

File: apps/windows/PasteSheet/Domain/Entities/PasteItem.cs
using PasteSheet.Data.DTOs;

namespace PasteSheet.Domain.Entities;

public sealed record PasteItem(long Id, string Content, string Directory, string CreatedAt, string? Memo)
{
    public PasteItem(PasteItemDTO dto)
        : this(dto.Id, dto.Content, dto.Directory, dto.CreatedAt, dto.Memo) { }
}
